### ISSUE TITLE
Thin

### ISSUE THREAD

**Issue by mfaughn** [sj26/skinny#7]
Restricting thin to be less than 1.7 is a real pain.  The mailcatcher gem is, by a huge margin, the main client of this gem and it has no such restriction on thin...but you're effectively imposing one for it.

Is there some known issue with versions of thin >= 1.7?

**Comment by sj26** [comment]
Yes, it doesn't work. If someone wanted to fix this so that it does, or switch to another websocket implementation, patches would be welcome. :-)

**Comment by sj26** [comment]
I'd probably switch mailcatcher to https://github.com/faye/websocket-driver-ruby using https://github.com/faye/faye-websocket-ruby, it's pretty similar from this example: https://github.com/Joe-noh/sinatra-faye-websocket-example/blob/master/app.rb
